pkg/client/query/sharedquerier.go

```diff
@@ -49,11 +53,19 @@ func NewSharedQuerier(deps depinject.Config) (client.SharedQueryClient, error) {
 // Once `ModuleParamsClient` is implemented, use its replay observable's `#Last()` method
 // to get the most recently (asynchronously) observed (and cached) value.
 func (sq *sharedQuerier) GetParams(ctx context.Context) (*sharedtypes.Params, error) {
+	// Get the params from the cache if they exist.
+	if params, found := sq.paramsCache.Get(); found {
```

Comment:
I'm concerned (and don't fully understand) the lack of a "height" param when retrieving things from the cache.

Reply:
This cache implementation does not add any new functionality besides caching whatever has been queried.

It does not alter the `RelayMiner`s current behavior
*  `RelayMiner` cold start
* React to `Param`s change

For those reasons, it does not leverage historical data that justifies the usage of `height` for cache querying.

Reply:
My understanding is that the cache implementations here are NOT historical; i.e. ONLY the most recently observed value is cached for each `ParamsCache` instance (or key, in the case of `KeyValueCache`).

While #994 does include historical caching as well (via the `HistoricalQueryCache` interface, that's an additional and distinct feature.

This shouldn't be necessary here because we're clearing the cache on every new block. The end result being, somewhat sub-optimal, but significant caching. This reformulates the number off-chain queries from being a function of API usage, to no more than one per block, per cache.